scripts/zones/Waughroon_Shrine/mobs/Titanis_Jax.lua

```diff
@@ -8,16 +8,73 @@ mixins = { require('scripts/mixins/job_special') }
 ---@type TMobEntity
 local entity = {}
 
+entity.onMobInitialize = function(mob)
+    mob:setMobMod(xi.mobMod.MAGIC_COOL, 20)
+end
+
 entity.onMobSpawn = function(mob)
+    mob:setLocalVar('spellList', 1)
+    mob:setMod(xi.mod.REGAIN, 250)
     xi.mix.jobSpecial.config(mob, {
         specials =
         {
-            { id = xi.jsa.SOUL_VOICE, cooldown = 200, hpp = 95 },
+            { id = xi.jsa.SOUL_VOICE, hpp = 60 },
         },
     })
 end
 
+entity.onMobMagicPrepare = function(mob, target, spellId)
+    local spellTable = {
+        [1] = {
+            xi.magic.spell.ARMYS_PAEON_IV,
+            xi.magic.spell.DARK_CAROL,
+            xi.magic.spell.EARTH_CAROL,
+            xi.magic.spell.FIRE_CAROL,
+            xi.magic.spell.ICE_CAROL,
+            xi.magic.spell.LIGHT_CAROL,
+            xi.magic.spell.LIGHTNING_CAROL,
+            xi.magic.spell.WATER_CAROL,
+            xi.magic.spell.WIND_CAROL,
+        },
+        [2] = {
+            xi.magic.spell.ARMYS_PAEON_V,
+            xi.magic.spell.DARK_CAROL,
+            xi.magic.spell.EARTH_CAROL,
+            xi.magic.spell.FIRE_CAROL,
+            xi.magic.spell.ICE_CAROL,
+            xi.magic.spell.LIGHT_CAROL,
+            xi.magic.spell.LIGHTNING_CAROL,
+            xi.magic.spell.WATER_CAROL,
+            xi.magic.spell.WIND_CAROL,
+        },
+    }
+    local spellList = spellTable[mob:getLocalVar('spellList')]
+    return spellList[math.random(1, #spellList)]
```

Comment:
```suggestion
    return utls.randomEntry(spellList)
```

small change for readability on all these dual-spell-lists. Also works for non-sequential tables, which is nice

Reply:
thanks for the suggestion, ill get this one implemented!

Reply:
Btw idk why the changes got undone but you may want to consider adjusting your process for updating PR branches

Reply:
I requested this changes to be undone. In this case, we really do not need to call a function, that itself calls another function which contains a loop, to not even save a single line of code.